src/js/view/popup/popupWrite.js

```diff
@@ -0,0 +1,372 @@
+/**
+ * @fileoverview Floating layer for writing new schedules
+ * @author NHN Ent. FE Development Team <dl_javascript@nhnent.com>
+ */
+'use strict';
+
+var View = require('../../view/view');
+var FloatingLayer = require('../../common/floatingLayer');
+var util = require('tui-code-snippet');
+var DatePicker = require('tui-date-picker');
+var config = require('../../config'),
+    domevent = require('../../common/domevent'),
+    domutil = require('../../common/domutil');
+var tmpl = require('../template/popup/popupWrite.hbs');
+var mmax = Math.max;
+var mmin = Math.min;
+
+/**
+ * @constructor
+ * @extends {View}
+ * @param {HTMLElement} container = container element
+ */
+function PopupWrite(container) {
+    View.call(this, container);
+
+    /**
+     * @type {FloatingLayer}
+     */
+    this.layer = new FloatingLayer(null, container);
+
+    /**
+     * cached view model
+     * @type {object}
+     */
+    this._viewModel = null;
+    this.tmpl = tmpl;
+    this.focusedDropdown = null;
+    this.currentCalendar = null;
+    this.currentState = null;
+    this.calendars = [];
+
+    domevent.on(container, 'click', this._onClick, this);
+}
+
+util.inherit(PopupWrite, View);
+
+/**
+ * Mousedown event handler for hiding popup layer when user mousedown outside of
+ * layer
+ * @param {MouseEvent} mouseDownEvent - mouse event object
+ */
+PopupWrite.prototype._onMouseDown = function(mouseDownEvent) {
+    var target = (mouseDownEvent.target || mouseDownEvent.srcElement),
+        popupLayer = domutil.closest(target, config.classname('.floating-layer'));
+
+    if (popupLayer) {
+        return;
+    }
+
+    this.hide();
+};
+
+/**
+ * @override
+ */
+PopupWrite.prototype.destroy = function() {
+    this.layer.destroy();
+    this.layer = null;
+    domevent.off(this.container, 'click', this._onClick, this);
+    domevent.off(document.body, 'mousedown', this._onMouseDown, this);
```

Comment:
Detaching `mousedown` event for `document.body` is duplicate at `hide()` function.